Content.Server/GameObjects/Components/WallStorage/WallStorageComponent.cs

```diff
@@ -0,0 +1,152 @@
+using Content.Shared.GameObjects.EntitySystems;
+using Content.Shared.Interfaces;
+using Robust.Server.GameObjects;
+using Robust.Shared.GameObjects;
+using Robust.Shared.Interfaces.GameObjects;
+using Robust.Shared.IoC;
+using Robust.Shared.Localization;
+
+namespace Content.Server.GameObjects.Components.WallStorage
+{
+    [RegisterComponent]
+    [ComponentReference(typeof(IActivate))]
+    [ComponentReference(typeof(IInteractUsing))]
+    public class WeaponStorageComponent : Component, IActivate, IInteractUsing
+    {
+        protected IEntity _heldItem;
+        protected ContainerSlot _container;
+        protected AppearanceComponent _appearanceComponent;
+        public WallStorageStatus Status => _status;
+        protected WallStorageStatus _status;
+
+        public override string Name => "WeaponStorage";
+
+        bool IInteractUsing.InteractUsing(InteractUsingEventArgs eventArgs)
+        {
+            var result = TryInsertItem(eventArgs.Using);
+            if (!result)
+            {
+                var localizationManager = IoCManager.Resolve<ILocalizationManager>();
+                eventArgs.User.PopupMessage(Owner, localizationManager.GetString("Unable to insert item"));
+            }
+
+            return result;
+        }
+
+        public void RemoveItem(IEntity user)
+        {
+            var heldItem = _container.ContainedEntity;
+            if (heldItem == null)
+            {
+                return;
+            }
+
+            _container.Remove(heldItem);
+            if (user.TryGetComponent(out HandsComponent handsComponent))
+            {
+                handsComponent.PutInHandOrDrop(heldItem.GetComponent<ItemComponent>());
+            }
+        }
+
+        void IActivate.Activate(ActivateEventArgs eventArgs)
+        {
+            RemoveItem(eventArgs.User);
+        }
+
+        [Verb]
+        private sealed class InsertVerb : Verb<WeaponStorageComponent>
+        {
+            protected override void GetData(IEntity user, WeaponStorageComponent component, VerbData data)
+            {
+                if (!ActionBlockerSystem.CanInteract(user))
+                {
+                    data.Visibility = VerbVisibility.Invisible;
+                    return;
+                }
+
+                if (!user.TryGetComponent(out HandsComponent handsComponent))
+                {
+                    data.Visibility = VerbVisibility.Invisible;
+                    return;
+                }
+
+                if (handsComponent.GetActiveHand == null)
+                {
+                    data.Visibility = VerbVisibility.Disabled;
+                    data.Text = Loc.GetString("Insert");
+                    return;
+                }
+
+                if (component._container.ContainedEntity != null)
+                {
+                    data.Visibility = VerbVisibility.Disabled;
+                }
+
+                data.Text = $"Insert {handsComponent.GetActiveHand.Owner.Name}";
```

Comment:
It's not very consistent in the codebase, but this should be localised (though I appreciate this was probably copied from somewhere).